src/mcp/tools/core/init.ts

```diff
@@ -4,136 +4,463 @@ import { toContent } from "../../util";
           ),
-        storage: z
-          .object({
-            rules_filename: z
-              .string()
-              .optional()
-              .default("storage.rules")
-              .describe("The file to use for Firebase Storage Security Rules."),
-            rules: z
-              .string()
-              .optional()
-              .describe(
-                "The security rules to use for Firebase Storage Security Rules. Default to closed rules that deny all access.",
-              ),
-          })
+      })
+      .optional()
+      .describe(
+        "Provide this object to initialize Firebase Data Connect with Cloud SQL Postgres in this project directory.\n" +
+          "It installs Data Connect Generated SDKs in all detected apps in the folder.",
+      ),
+    storage: z
+      .object({
+        rules_filename: z
+          .string()
+          .optional()
+          .default("storage.rules")
+          .describe("The file to use for Firebase Storage Security Rules."),
+        rules: z
+          .string()
           .optional()
           .describe(
-            "Provide this object to initialize Firebase Storage in this project directory.",
+            "The security rules to use for Firebase Storage Security Rules. Default to closed rules that deny all access.",
           ),
-      }),
-    }),
+      })
+      .optional()
+      .describe("Provide this object to initialize Firebase Storage in this project directory."),
+    ai_logic: z
+      .boolean()
+      .optional()
+      .describe("Enable Firebase AI Logic feature (requires provisioning to be enabled)"),
+  }),
+});
+
+type ProvisioningInput = z.infer<typeof inputSchema>["provisioning"];
+type ProjectInput = z.infer<typeof inputSchema>["project"];
+type AppInput = z.infer<typeof inputSchema>["app"];
+
+export const init = tool(
+  {
+    name: "init",
+    description:
+      "Initializes selected Firebase features in the workspace (Firestore, Data Connect, Realtime Database, Firebase AI Logic). All features are optional; provide only the products you wish to set up. " +
+      "You can initialize new features into an existing project directory, but re-initializing an existing feature may overwrite configuration. " +
+      "To deploy the initialized features, run the `firebase deploy` command after `firebase_init` tool.",
+    inputSchema: inputSchema,
     annotations: {
       title: "Initialize Firebase Products",
       readOnlyHint: false,
       idempotentHint: true,
     },
     _meta: {
       requiresProject: false, // Can start from scratch.
-      requiresAuth: false, // Will throw error if the specific feature needs it.
+      requiresAuth: true, // Required for provisioning and Firebase operations.
     },
   },
-  async ({ features }, { projectId, config, rc }) => {
+  async ({ features, provisioning, project, app }, { projectId, config, rc }) => {
+    validateProvisioningInputs(provisioning, project, app);
+
+    if (provisioning?.enable) {
```

Comment:
Haven't looked in-depth yet.

High-level thought is the orchestration API specific logic should be pulled out to a separate package.

Similar to [commands/*](https://github.com/firebase/firebase-tools/tree/master/src), those are entry point to MCP tools. We should keep it focused on defining the input argument of the tool and delegate to other modules for more involved logics.